viewing_party/party.py

```diff
@@ -0,0 +1,204 @@
+import operator
+
+# Wave 1 
+def create_movie(title, genre, rating):
+    if title and genre and rating:
+        movie_to_watch = {
+                "title" : title,
+                "genre" : genre,
+                "rating" : rating 
+                }    
+        return movie_to_watch      
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    list_of_watchlist = user_data["watchlist"]
+    for movie_to_watch in list_of_watchlist:
+        if movie_to_watch["title"] == title:
+            add_to_watched(user_data, movie_to_watch)
+            list_of_watchlist.remove(movie_to_watch)
+            return user_data
+    return user_data
+
+# Wave 2 
+def get_watched_avg_rating(user_data):
+    sum_rating = 0 
+    avg_rating = 0 
+    list_of_watched = user_data["watched"]
+    if list_of_watched:
+        for movie_watched in list_of_watched:          
+            sum_rating += movie_watched["rating"]
+        avg_rating = sum_rating / len(list_of_watched)
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    list_of_watched = user_data["watched"]
+    if list_of_watched:
+        dict_genre_to_count = {}
+        for movie_watched in list_of_watched:
+            movie_watched_genre = movie_watched["genre"] 
+            if movie_watched_genre in dict_genre_to_count:
+                dict_genre_to_count[movie_watched_genre] = dict_genre_to_count.get(movie_watched_genre) + 1
+            else:
+                dict_genre_to_count[movie_watched_genre] = 1
+
+    # get the most genre 
+        max_count_genre = max(dict_genre_to_count, key = dict_genre_to_count.get)
+        return max_count_genre
+        
+    #  Alternative approach 
+        # cur_most_genre = list_of_watched[0]["genre"] # this is a var to store temp value
+        # count_cur_most_genre = dict_genre_to_count[cur_most_genre]
+        # for key_value_genre_count in dict_genre_to_count:
+        #     if key_value_genre_count.values() > count_cur_most_genre:
+        #         count_cur_most_genre = key_value_genre_count.values()
+        
+        # # get the key
+        # for key_genre in dict_genre_to_count:
+        #     if dict_genre_to_count[key_genre] == count_cur_most_genre:
+        #         return key_genre
+
+        # how about if there are more than one genre with the same most count?
+        
+
+# Wave 3
+def get_unique_watched(user_data):
+    list_unique_movie_user_watched = []
+    set_user_watched = set()
+    set_friends_watched = set()
+
+    # to add movie title in set_user_watched
+    helper_func_add_movie_title_to_set(set_user_watched,user_data)
+
+    # to add movie title in set_friends_watched
+    for friends_data in user_data["friends"]:
+        helper_func_add_movie_title_to_set(set_friends_watched,friends_data)
```

Comment:
I like the use of helper functions that you created. Thank you for sharing this to students in class